src/handshake/client.rs

```diff
@@ -42,9 +43,7 @@ pub struct Client<'a, T> {
     /// The HTTP origin header.
     origin: Option<&'a str>,
     /// A buffer holding the base-64 encoded request nonce.
-    nonce: [u8; 32],
```

Comment:
do you know why it was 32 before?

Reply:
I think it was just to be safe, the nonce was always sliced to the length given by base64 encoding, I've skipped it altogether here since it's guaranteed to be 24 (per comment on `WebSocketKey`).